### ISSUE TITLE
jTracert fails when parent class is loaded by bootstrap or extension classloader and throws an exception

### ISSUE THREAD

**Issue by GoogleCodeExporter** [ankit4u3/jtracert#11]
```
What steps will reproduce the problem?

jTracert fails when parent class is loaded by bootstrap or extension
classloader and throws an exception.
You can use following application to reproduce this defect:

package com.google.code.jtracert.samples;

import java.util.Map;
import java.util.Set;
import java.util.Collection;
import java.util.HashMap;

class ExpectedException extends RuntimeException {

}

class FailMap implements Map {

    private void fail() {
        throw new ExpectedException();
    }

    public int size() {
        fail();
        return 0;
    }

    public boolean isEmpty() {
        fail();
        return false;
    }

    public boolean containsKey(Object key) {
        fail();
        return false;
    }

    public boolean containsValue(Object value) {
        fail();
        return false;
    }

    public Object get(Object key) {
        fail();
        return null;
    }

    public Object put(Object key, Object value) {
        fail();
        return null;
    }

    public Object remove(Object key) {
        fail();
        return null;
    }

    public void putAll(Map t) {
        fail();

    }

    public void clear() {
        fail();

    }

    public Set keySet() {
        fail();
        return null;
    }

    public Collection values() {
        fail();
        return null;
    }

    public Set entrySet() {
        fail();
        return null;
    }
}

class CustomMap extends HashMap {

    @SuppressWarnings("unchecked")
    public CustomMap(Map m) {
        super(m);
    }

}

public class BootStrapClassConstructor {

    @SuppressWarnings("unchecked")
    public static void main(String[] arguments) throws Exception {
        try {
            Map map = new FailMap();
            Map adapterMap = new CustomMap(map);
        } catch (ExpectedException e) {

        }
    }

}

It's added to the integration tests as 'bootStrapClassConstructor'

What is the expected output? What do you see instead?

Method call trace should be generated.
Instead there's a null pointer exception:
    [junit]     at
com.google.code.jtracert.traceBuilder.impl.MethodCallTraceBuilderImpl.leaveConst
ructor(MethodCallTraceBuilderImpl.java:432)
    [junit]     at com.google.code.jtracert.samples.CustomMap.<init>(Unknown
Source)
    [junit]     at
com.google.code.jtracert.samples.BootStrapClassConstructor.main(Unknown Source)



In order to fix this defect we must instrument the constructors of all
classes (even loaded by bootstrap and extensions classloaders). Suppose we
can exclude final classes.

Classes loaded by bootstrap classloader should call synthetic methods added
to java.lang.System class


```

Original issue reported on code.google.com by `Dmitry.Bedrin` on 24 Feb 2009 at 4:37

**Comment by GoogleCodeExporter** [comment]
```
Fixed in SVN revision r175
```

Original comment by `Dmitry.Bedrin` on 4 Mar 2009 at 1:01
- Changed state: **Fixed**

**Comment by GoogleCodeExporter** [comment]
```
Fixed in jTracert version 0.1.0
```

Original comment by `Dmitry.Bedrin` on 17 Mar 2009 at 10:12
